cookbook/docs/getting_started/package_register.md

```diff
@@ -0,0 +1,350 @@
+One way it achieves this is by providing a way for you to bake in user-defined
+workflows and all of their dependencies into a Docker container.
+
+The third method of registering your workflows uses two commands:
+
+- `pyflyte package`: packages your tasks and workflows into protobuf format.
+- `flytectl register`: registers the Flyte package to the configured cluster.
+
+This is the production-grade registration flow that we recommend because this
+method ensures that the workflows are fully containerized, which ensures that
+the system- and Python-level dependencies along with your workflow source code
+are immutable.
+
+
+**Containerizing your Project**
+
+Flyte relies on Docker to containerize your code and third-party dependencies.
+When you invoke `pyflyte init`, the resulting template project ships with a
+`docker_build.sh` script that you can use to build and tag a container according
+to the recommended practice:
+
+```{prompt} bash $
+./docker_build.sh
+```
+
+```{important}
+By default, the `docker_build.sh` script:
+
+- Uses the `PROJECT_NAME` specified in the `pyflyte init` command, which in
+  this case is `my_project`.
+- Will not use any remote registry.
+- Uses the git sha to version your tasks and workflows.
+```
+
+You can override the default values with the following flags:
+
+```{prompt} bash $
+./docker_build.sh -p <PROJECT_NAME> -r <REGISTRY> -v <VERSION>
+```
+
+For example, if you want to push your Docker image to Github's image registry
+you can specify the `-r ghcr.io` flag.
+
+```{note}
+The `docker_build.sh` script is purely for convenience; you can always roll
+your own way of building Docker containers.
+```
+
+**Package your Project with `pyflyte package`**
+
+You can package your project with the `pyflyte package` command like so:
+
+```{prompt} bash $
+pyflyte --pkgs workflows package --image ghcr.io/flyteorg/flytekit:py3.9-latest
+```
+
+````{div} shadow p-3 mb-8 rounded
+
+**Expected Output:**
+
+```{code-block} bash
+Successfully serialized 4 flyte objects
+  Packaging workflows.example.say_hello -> 0_workflows.example.say_hello_1.pb
+  Packaging workflows.example.greeting_length -> 1_workflows.example.greeting_length_1.pb
+  Packaging workflows.example.wf -> 2_workflows.example.wf_2.pb
+  Packaging workflows.example.wf -> 3_workflows.example.wf_3.pb
+Successfully packaged 4 flyte objects into /Users/nielsbantilan/sandbox/my_project/flyte-package.tgz
+```
+
+````
+
+This will create a portable package `flyte-package.tgz` containing all the Flyte
+entities compiled as protobuf files that you can register with multiple Flyte
+clusters.
+
+**Register with `flytectl register`**
+
+Finally, register your tasks and workflows with `flytectl register files`:
+
+```{prompt} bash $
+flytectl register files \
+    --project flytesnacks \
+    --domain development \
+    --archive flyte-package.tgz \
+    --version "$(git rev-parse HEAD)"
+```
+
+```{note}
```

Comment:
This callout should be converted into a normal section.

Reply:
for feedback like this it would be useful to have a rationale.